src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.RemoveUnusedVariable;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.OrganizeImports;
+using Microsoft.CodeAnalysis.RemoveUnnecessaryImports;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Editor.CodeCleanup;
+using Microsoft.VisualStudio.Language.CodeCleanUp;
+
+namespace Microsoft.VisualStudio.LanguageServices.CSharp.LanguageService
+{
+    internal class CSharpCodeCleanupFixer : ForegroundThreadAffinitizedObject, ICodeCleanUpFixer
+    {
+        public const string RemoveUnusedImports = nameof(RemoveUnusedImports);
+        public const string SortImports = nameof(SortImports);
+
+        private readonly ICodeFixService _codeFixServiceOpt;
+        /// <summary>
+        /// TODO: hardcoded list need to be replace by inclusion/exclusion list from .editorconfig
+        /// </summary>
+        private ImmutableArray<string> _errorCodes = ImmutableArray.Create(RemoveUnusedImports, 
```

Comment:
isn't this just duplicates of ones in code clean up service already?

Reply:
ah. is this new editor feature?

Reply:
so. we don't follow options user has set?

Reply:
it is a new code cleanup command (Ctrl + E, C) and the options are set in .editorconfig

Reply:
shouldn't we share that option with code clean up service?

Reply:
➡️ The code clean up service is getting removed in a separate pull request for 16.0 preview 1.